src/game.cpp

```diff
@@ -3022,8 +3022,7 @@ bool game::handle_action()
                     }
                     as_m.text = data.str();
                 }
-                if( (u.has_item_with_flag("ALARMCLOCK") || u.has_bionic("bio_watch")) &&
-                    !(u.hunger < -60) ) {
+                if  ( u.has_alarm_clock() && !(u.hunger < -60) ) {
```

Comment:
This check excludes the bionic Internal Chronometer, previously that bionic allowed the player to use the alarm function, now one needs an external item with that flag.

If this is the desired state, somebody should update the description of the bionic because it still says "You have been equipped with an internal atomic clock, ensuring that you will always know the current time.  **Additionally, it includes a silent alarm clock function.**"

Or just add a check for the bionic to `player::has_alarm_clock` to get the previous behavior back.